rootfs/etc/nginx/lua/balancer/ewma.lua

```diff
@@ -22,28 +49,51 @@ local function decay_ewma(ewma, last_touched_at, rtt, now)
   return ewma
 end
 
-local function get_or_update_ewma(self, upstream, rtt, update)
-  local ewma = self.ewma[upstream] or 0
+local function get_or_update_ewma(upstream, rtt, update)
+  local lock_err = nil
+  if update then
+    lock_err = lock(upstream)
+  end
+  local ewma = ngx.shared.balancer_ewma:get(upstream) or 0
+  if lock_err ~= nil then
+    return ewma, lock_err
```

Comment:
I'd like to note that `lock_err` gets ignored. In our DC implementation we also ignore this, but the difference is we emit statsd metric when an error happens.